src/libraries/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -327,13 +373,13 @@ public unsafe BitArray And(BitArray value)
                 case 0: goto Done;
             }
 
-            int i = 0;
+            uint i = 0;
             if (Avx2.IsSupported)
```

Comment:
Not something we should do in this PR, but you may notice that the x86 and ARM sides are essentially identical, aside from the ISA queried.

It would likely be beneficial if we abstracted some of the shared functionality somewhere so we don't have essentially two identical code paths. (Pending codegen optimizing everything correctly, etc).